### PR TITLE
re-enable a cypress test, removing use of aliases

### PR DESCRIPTION
**Pull request by aitchiss**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [X] Optimization
- [ ] Documentation Update

## Description

We've been experiencing a lot of flake in autocomplete tests, and currently have two main offenders skipped. @nickytonline and I had a look into these and we think the main problem is related to aliasing and the fact the autocomplete component works in a bit of an unusual way, with different textareas appearing and disappearing.

This PR reintroduces _one_ of the skipped tests, with a new approach that avoids all usage of aliasing, and adds a few more intermediary checks to help make sure elements have fully appeared before interacting with them.

This PR however doesn't rework any of the tests that weren't already skipped (other than to target the full 'option' element in the dropdown, instead of relying on 'findByText'). If this fix proves stable once merged, we can bring the other tests over to the same approach, making them more robust.

## Related Tickets & Documents

N/A

## QA Instructions, Screenshots, Recordings

As of this moment, I have re-run the cypress tests Travis job 10 times, and all have been successful. I'll retrigger it a few more times later today, and if it keeps reliably passing, then I think we are good.

### UI accessibility concerns?

N/A

## Added tests?

- [X] Yes - updated test
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [X] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

I don't trust you

### CODE REVIEW

File: cypress/integration/articleFlows/commentOnArticle.spec.js [hunk 1]
```diff
@@ -402,3 +404,15 @@ describe('Comment on articles', () => {
     });
   });
 });
+
+const getCommentCombobox = () =>
```
Comment (nickytonline) [inline cypress/integration/articleFlows/commentOnArticle.spec.js:408]: Documentation (blocking): Leave a comment in the code as to why we are not using aliases and these functions instead.

nitpick (non-blocking): Move these functions to the top of the file within the first describe.

**Review by nickytonline — CHANGES_REQUESTED** [review]
I'm getting a test failure for the last test.

image

Also, a small request change and a suggestion.

**Comment by aitchiss** [comment]
> I'm getting a test failure for the last test.

@nickytonline that test isn't one that's currently disabled or looked at in this PR. Maybe we want to consider skipping _all_ of the mention autocomplete tests (apart from the first one we're reintroducing here) and bringing them back in one by one when we're convinced they're stable?

I don't want to just disable everything, but at the same time I think the plan of bringing this one test back and then rolling out the approach to the other tests once we're sure that it's stable makes sense. I haven't seen that particular test flaking locally for me in CI, but clearly it's not super reliable.

**Comment by nickytonline** [comment]
> > I'm getting a test failure for the last test.
> 
> @nickytonline that test isn't one that's currently disabled or looked at in this PR. Maybe we want to consider skipping _all_ of the mention autocomplete tests (apart from the first one we're reintroducing here) and bringing them back in one by one when we're convinced they're stable?
> 
> I don't want to just disable everything, but at the same time I think the plan of bringing this one test back and then rolling out the approach to the other tests once we're sure that it's stable makes sense. I haven't seen that particular test flaking locally for me or in CI, but clearly it's not super reliable.

Ahh, didn't realize that one wasn't part of your changes. 🙃 It could be something on my local. If it's been reliable on CI, let's not worry for the moment and stick to the rollout out we discussed of unflaking one test at time until it's stable for a few days. 😎

So just the small request change and this is good to go then.

**Comment by aitchiss** [comment]
@nickytonline added that comment, and removed aliases from the other enabled tests in that file 🙌

**Review by nickytonline — APPROVED** [review]
LGTM! Hopefully we can say adieu to this flakiness. 😎

**Review by rhymes — APPROVED** [review]
LGTM!

**Pull request merged**